icefall/decode.py

```diff
@@ -854,11 +855,21 @@ def rescore_with_attention_decoder(
         sos_id=sos_id,
         eos_id=eos_id,
     )
+
     assert nll.ndim == 2
     assert nll.shape[0] == len(token_ids)
 
     attention_scores = -nll.sum(dim=1)
 
+    if nnlm_evaluator is not None:
+        aux_labels = k2.RaggedTensor(tokens_shape, nbest.fsa.aux_labels)
+        aux_labels = aux_labels.remove_values_leq(0)
+        aux_labels = aux_labels.tolist()
```

Comment:
Eventually it would be nice if we can make more use of ragged tensors and less of lists (more efficient).  But I'd say this is very low priority.